src/mlpack/methods/random_forest/bootstrap.hpp

```diff
@@ -18,30 +18,229 @@ namespace mlpack {
-         typename WeightsType>
-void Bootstrap(const MatType& dataset,
-               const LabelsType& labels,
-               const WeightsType& weights,
-               MatType& bootstrapDataset,
-               LabelsType& bootstrapLabels,
-               WeightsType& bootstrapWeights)
+class DefaultBootstrap
 {
-  bootstrapDataset.set_size(dataset.n_rows, dataset.n_cols);
-  bootstrapLabels.set_size(labels.n_elem);
-  if (UseWeights)
-    bootstrapWeights.set_size(weights.n_elem);
+ public:
+  template<bool UseWeights,
+           typename MatType,
+           typename LabelsType,
+           typename WeightsType>
+  void Bootstrap(const MatType& dataset,
+                 const LabelsType& labels,
+                 const WeightsType& weights,
+                 MatType& bootstrapDataset,
+                 LabelsType& bootstrapLabels,
+                 WeightsType& bootstrapWeights)
+  {
+    bootstrapDataset.set_size(dataset.n_rows, dataset.n_cols);
+    bootstrapLabels.set_size(labels.n_elem);
+    if (UseWeights)
+      bootstrapWeights.set_size(weights.n_elem);
 
-  // Random sampling with replacement.
-  arma::uvec indices = randi<arma::uvec>(dataset.n_cols,
+    // Random sampling with replacement.
+    arma::uvec indices = randi<arma::uvec>(dataset.n_cols,
       arma::distr_param(0, dataset.n_cols - 1));
-  bootstrapDataset = dataset.cols(indices);
-  bootstrapLabels = labels.cols(indices);
-  if (UseWeights)
-    bootstrapWeights = weights.cols(indices);
-}
+    bootstrapDataset = dataset.cols(indices);
+    bootstrapLabels = labels.cols(indices);
+    if (UseWeights)
+      bootstrapWeights = weights.cols(indices);
+  }
+};
+
+/**
+ * No-op bootstrap strategy that just copies the input to the output.
+ */
+class IdentityBootstrap
+{
+ public:
+  template<bool UseWeights,
+           typename MatType,
+           typename LabelsType,
+           typename WeightsType>
+  void Bootstrap(const MatType& dataset,
+                 const LabelsType& labels,
+                 const WeightsType& weights,
+                 MatType& bootstrapDataset,
+                 LabelsType& bootstrapLabels,
+                 WeightsType& bootstrapWeights)
+  {
+    MakeAlias(bootstrapDataset, dataset, dataset.n_rows, dataset.n_cols);
+    MakeAlias(bootstrapLabels, labels, labels.n_elem);
+    if (UseWeights)
+      MakeAlias(bootstrapWeights, weights, weights.n_elem);
+  }
+};
+
+/**
+ * Sequential bootstrap.
+ * 
+ * This sequential bootstrap is suitable for time-series that are not
+ * independent and identically distributed (IID). The bootstrap will
+ * generate variations of the data that are more IID. In order to do
+ * that it takes an indicator matrix that is a measure for how
+ * much informational overlap exists between each data point in the
+ * data set. Then it will randomly draw from the dataset with
+ * replacement but reducing the likelihood to draw data points that
+ * have high informational overlap with already drawn samples.
```

Comment:
Do you want to update this comment?  Specifically we take an `intervals` matrix now instead of an indicator matrix.